pages/deploy/deploy_aws.md

```diff
@@ -315,6 +329,85 @@ deploy to 2 deployments accounts.
+    "@aws-cdk/core:stackRelativeExports": false,
+    "tooling_vpc_id": "vpc-1234567890EXAMPLE",
+    "tooling_region": "eu-west-2",
+    "tooling_vpc_restricted_nacl": true,
+    "git_branch": "master",
+    "git_release": true,
+    "quality_gate": false,
+    "resource_prefix": "da",
+    "DeploymentEnvironments": [
+        {
+            "envname": "dev",
+            "account": "000000000000",
+            "region": "eu-west-1",
+            "with_approval": false,
+            "internet_facing": true,
+            "prod_sizing": false,
+             "custom_auth": {
+               "provider": "okta",
+               "url": "https://URL",
+               "redirect_url": "https://redirect_url.net",
+               "client_id": "091sdz02308042",
+               "response_types": "code",
+               "scopes": "openid profile",
+               "jwks_url": "https://jwksurl",
+               "claims_mapping": {
+                 "user_id": "id",
+                 "email": "user_email"
+               }
+             }
+        }
+    ]
+  }
+}
+```
+
+### 6.1 (Important) - Configure data.all to use external Idp
+
+Please read this section only if you are deploying data.all with an external Idp.
+
+Out of the box, data.all uses Cognito as the Idp and the user pool provider. Data.all can also be configured to use an external Idp
+which supports OIDC. In order to do so, please add `custom_auth` section into your `cdk.json` file as shown in **Example-3**. 
+
+This will take care of configuring data.all frontend to use the external Idp URL at the time of login. Also, it will setup a 
+custom authorizer lambda and attach it to the API gateway. This custom authorizer will lambda will work in place of the cognito authorizer,
+which is deployed by default. 
+
+**Important** - 
+When setting up data.all, the user pool is managed by Cognito. Here you can make teams and groups users in a team. This forms the data.all team.
+When using custom authentication, this information will have to be maintained by either your Idp or any other systems you use to maintain
+group information. 
+
+When using custom authentication, you will have to create a file and implement 3 functions from dataall/base/services/ServiceProvider.py
+For example, you can create a package like dataall/base/custom_auth/MyProvider.py and create your own class and implement the methods as shown below
+```python
+class MyProvider(ServiceProvider):
+    def get_user_emailids_from_group(self, role_name):
+        # Your implementation
+    
+    def get_groups_for_user(self, user_id):
+        # Your implementation
+    
+    def list_groups(self, envname: str, region: str):
+        # Your implementation
+```
+
+Once you implement these methods, create an instance of your provider in dataall/base/service/ServiceProviderFactory.py. Please follow the 
+instructions in the comments to create and return and instance of your custom service provider.
+
```

Comment:
Maybe we can add here a small code snippet example

Reply:
Added . Thanks !